internal/services/aws_reservation_service.go

```diff
@@ -138,17 +157,30 @@ func CreateAWSReservation(w http.ResponseWriter, r *http.Request) {
 			AccountID:     accountId,
 			ReservationID: reservation.ID,
 			PubkeyID:      pk.ID,
-			Name:          reservation.Name,
+			Detail:        reservation.Detail,
 			AMI:           ami,
 			ARN:           arn,
-			Amount:        reservation.Amount,
-			InstanceType:  reservation.InstanceType,
-			PowerOff:      reservation.PowerOff,
 		},
 	}
 
+	startJobs := []dejq.PendingJob{uploadPubkeyJob, launchJob}
+
+	// Optionally, rename instances
+	if doRenameInstances {
+		logger.Debug().Msgf("Enqueuing rename instance job for source %s", reservation.SourceID)
+		renameJob := dejq.PendingJob{
+			Type: queue.TypeRenameInstanceAws,
+			Body: &jobs.RenameInstanceJobArgs{
+				AccountID:     accountId,
+				ReservationID: reservation.ID,
+				Detail:        reservation.Detail,
+			},
+		}
+		startJobs = append(startJobs, renameJob)
+	}
+
 	// Enqueue all jobs
-	err = queue.GetEnqueuer().Enqueue(r.Context(), uploadPubkeyJob, launchJob)
+	err = queue.GetEnqueuer().Enqueue(r.Context(), startJobs...)
 	if err != nil {
 		renderError(w, r, payloads.NewEnqueueTaskError(r.Context(), "AWS reservation", err))
 		return
```

Comment:
This change makes sure to schedule 2 jobs in case there are no names provided, otherwise it schedules 3 jobs (rename instances as the third one).